pkg/daemon/ceph/client/config.go

```diff
@@ -133,6 +140,28 @@ func generateConfigFile(context *clusterd.Context, clusterInfo *ClusterInfo, pat
 	return filePath, nil
 }
 
+func mergeDefaultConfigWithRookConfigOverride(context *clusterd.Context, clusterInfo *ClusterInfo, configFile *ini.File) error {
+	cm, err := context.Clientset.CoreV1().ConfigMaps(clusterInfo.Namespace).Get(k8sutil.ConfigOverrideName, metav1.GetOptions{})
+	if err != nil {
+		if !kerrors.IsNotFound(err) {
+			return errors.Wrapf(err, "failed to read configmap %q", k8sutil.ConfigOverrideName)
+		}
+		return nil
+	}
+
+	config, ok := cm.Data["config"]
+	if !ok {
+		logger.Debugf("No ceph configuration override to merge as %q configmap is empty", k8sutil.ConfigOverrideName)
```

Comment:
Actually, the message is not really accurate, `if !ok`, just means the key does not exist but it could be empty if present.